src/context/AuthContext.tsx

```diff
@@ -71,36 +88,41 @@ type AuthenticationProviderProps = {
   children: React.ReactNode
 }
 
-const projectNameForProxy = `@${Constants.expoConfig.owner}/${Constants.expoConfig.slug}`
-
 function useGoogleAuth() {
-  const [, response, promptAsync] = Google.useIdTokenAuthRequest(
-    {
-      expoClientId: clientId,
-      clientId,
-      androidClientId,
-      iosClientId,
-    },
-    {
-      projectNameForProxy,
+  const { execute: login, result } = useAsyncCallback(async () => {
+    try {
+      await GoogleSignin.hasPlayServices()
+      const response = await GoogleSignin.signIn()
+      if (GS.isSuccessResponse(response)) {
+        return { userInfo: response.data }
+      } else {
+        console.log('Google sign in cancelled')
+      }
+    } catch (error) {
+      if (GS.isErrorWithCode(error)) {
+        switch (error.code) {
+          case GS.statusCodes.IN_PROGRESS:
+            console.log('Google auth already in progress')
+            break
+          case GS.statusCodes.PLAY_SERVICES_NOT_AVAILABLE:
+            console.log('Google play services not available')
+            break
+          default:
+            console.log('Google sign in error', error)
+        }
+      } else {
+        console.log('An unexpected error occurred,', error)
```

Comment:
Instead of just logging errors, this should be surfaced to the user in an alert, toast or notification popup.

Reply:
Done